sqlmesh/core/state_sync/engine_adapter.py

```diff
@@ -564,11 +714,18 @@ def _push_snapshot_intervals(
         new_intervals = []
         for snapshot in snapshots:
             logger.info("Pushing intervals for snapshot %s", snapshot.snapshot_id)
+            cached_snapshot = self._snapshot_cache.get(snapshot.snapshot_id)
+            if not cached_snapshot and isinstance(snapshot, Snapshot):
+                cached_snapshot = self._snapshot_cache[snapshot.snapshot_id] = snapshot
             for start_ts, end_ts in snapshot.intervals:
+                if cached_snapshot:
+                    cached_snapshot.add_interval(start_ts, end_ts, is_dev=False)
```

Comment:
So if the given instance is `Snapshot` we'd add its own intervals to it? And cause merge again?